### PR TITLE
Fix PAX extended attribute reading logic to treat '=' character as valid in the value strings.

### LINKED ISSUE #81699: Pax Headers incorrectly treated is invalid if value contains an '=' sign
### Description

The current TarReader treats any extended headers which contain an equals sign as invalid. While I am not sure about the exact spec, it is in conflict with the [`MSWINDOWS.rawsd` extension](https://github.com/buildpacks/rfcs/blob/main/text/0076-windows-security-identifiers.md) which is used to define the Windows Security Descriptors for files. They are supposed to be encoded as base64 which can have `=` signs as padding characters. 

### Reproduction Steps

```cs
using System.Formats.Tar;

using var data = new MemoryStream();
using (var writer = new TarWriter(data, TarEntryFormat.Pax, leaveOpen: true))
{
    writer.WriteEntry(new PaxTarEntry(TarEntryType.RegularFile, "file.txt", new Dictionary<string, string>
    {
        ["MSWINDOWS.rawsd"] = "AQAAgBQAAAAkAAAAAAAAAAAAAAABAgAAAAAABSAAAAAhAgAAAQIAAAAAAAUgAAAAIQIAAA=="
    })
    {
        DataStream = new MemoryStream("Hello Pax"u8.ToArray())
    });
}

data.Position = 0;
using (var reader = new TarReader(data, leaveOpen: true))
{
    var entry = (PaxTarEntry)reader.GetNextEntry();
    if (!entry.ExtendedAttributes.TryGetValue("MSWINDOWS.rawsd", out var descriptor))
    {
        descriptor = "missing!";
    }
    
    Console.WriteLine("Security Descriptor: {0}", descriptor);
}
```

### Expected behavior

Output should show `Security Descriptor: AQAAgBQAAAAkAAAAAAAAAAAAAAABAgAAAAAABSAAAAAhAgAAAQIAAAAAAAUgAAAAIQIAAA==`

### Actual behavior

Output shows `Security Descriptor: missing!`

### Regression?

No, was like this since the beginning. 

### Known Workarounds

None known. 

### Configuration

.net Version: 7.0.102 

```
Edition	Windows 11 Enterprise
Version	22H2
Installed on	‎25/‎01/‎2023
OS build	22621.1105
Experience	Windows Feature Experience Pack 1000.22638.1000.0
```

Architecture: x64

### Other information

It currently seems by design to treat equals sign values as invalid even though it shouldn't be a problem to support it. From the remaining parsing code it seems to be expected that properties are separated by newlines and key-value by the first equals sign. 

Related code: https://github.com/dotnet/runtime/blob/167112574ebfcfcddfc23267e1ba93eebdcdeaf1/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs#L736-L740

I was extending https://github.com/dotnet/sdk-container-builds with features to support Windows Containers better and ran into this problem while creating unit tests to verify that the extended attributes are written.

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/runtime/issues/81699

We had logic in TarHeader.Read.cs that was explicitly forbidding the '=' character as part of the body of the value string in an extended attribute key-value pair. It felt natural to have such restriction in place as it is one of the two separators that the spec indicates:

- The '=' character is used to separate a key from a value.
- The '\n' character is used to separate a key-value-pair from another.

But the restriction is not needed: We only need to find the first '=' character to stop reading a key and start reading a value, and after that, we can keep reading any character, including any number of '=' characters, until we detect a '\n' character. Additionally, none of the specs indicated that the '=' is forbidden in values.

This issue was reported by a customer, so we would like to consider a backport to 7.0.

### CODE REVIEW

**Comment by carlossanlop** [comment]
While working on this, I realized we should disallow adding key-value-pairs in which the key contains '=', or in which either the key or the value contain a '\n'. Both cases will cause the extended attribute reading logic to fail.

I'll add the fixes to this PR unless anyone indicates a preference to have the two issues fixed separately.

**Comment by jeffhandley** [comment]
Separate sounds good to me. Would those other changes potentially be breaking?

**Review by jeffhandley — APPROVED** [review]

**Comment by carlossanlop** [comment]
> Would those other changes potentially be breaking?

I don't think adding the exception would be breaking because we would be shielding the user from unexpected behavior and corrupted data due to improper use of reserved characters in the key or values.

File: src/libraries/System.Formats.Tar/tests/TarTestsBase.Pax.cs [hunk 1]
```diff
@@ -124,5 +124,17 @@ protected void VerifyExtendedAttributeTimestamps(PaxTarEntry pax)
...
+        {
+            yield return new object[] { "key", "value" };
+            yield return new object[] { "key    ", "value    " };
+            yield return new object[] { "    key", "    value" };
+            yield return new object[] { "    key   ", "    value    " };
+            yield return new object[] { "    key spaced   ", "    value spaced    " };
+            yield return new object[] { "many sla/s\\hes", "/////////////\\\\\\///////////" };
+            yield return new object[] { "key", "va=lue" };
+            yield return new object[] { "key", "=" };
+        }
```
Comment (Jozkee) [inline src/libraries/System.Formats.Tar/tests/TarTestsBase.Pax.cs:138]: You may want to include the following test cases:
```
{ "key", "==" } // double equals sign
{ "key", "=foo" } // leading equals sign
{ "key", "foo=" } // trailing equals sign
```

**Review by jozkee — APPROVED** [review]

**Pull request merged**